### PR TITLE
Add removeTransaction implementation with test coverage

### PR DESCRIPTION
**Pull request by MahmoodTarek**
Implement and test removeTransaction functionality

- Added implementation for removeTransaction in InMemoryTransactionDataSourceImpl
- Created RemoveChecker.kt to validate different test cases for transaction removal
- Ensured edge cases are handled: non-existent transaction, empty list, etc.
- Used 'check' function for readable test output using "when...then..." format

### CODE REVIEW

File: Personal-Finance-Tracker/src/main/kotlin/test/TransactionCreationTest.kt
Comment (AbdElrahman-Rafaat-Amer) [inline Personal-Finance-Tracker/src/main/kotlin/test/TransactionCreationTest.kt:1]: If there are no changes in the file so prefer to not commit it.

File: Personal-Finance-Tracker/src/main/kotlin/Main.kt
Comment (AbdElrahman-Rafaat-Amer) [inline Personal-Finance-Tracker/src/main/kotlin/Main.kt:1]: If there are no changes in the file so prefer to not commit it.

File: Personal-Finance-Tracker/src/main/kotlin/test/RemoveTester.kt
Comment (AbdElrahman-Rafaat-Amer) [inline Personal-Finance-Tracker/src/main/kotlin/test/RemoveTester.kt:1]: I think it's better to rename this class to the name we agreed on (ex: TransactionRemoveTest).

File: Personal-Finance-Tracker/src/main/kotlin/test/RemoveTester.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    check(
+        name = "When removing same data but different ID, then should return false",
+        result =  false ,
+        expected =  false
+    )
+    check(
+        name = "When removing one of multiple transactions, then should return true",
+        result= true,
+        expected = true
+    )
```
Comment (ALiyASSER0) [inline Personal-Finance-Tracker/src/main/kotlin/test/RemoveTester.kt]: I think we need to add a case for when trying to delete an item that doesn't exist.

**Review by AbdElrahman-Rafaat-Amer — COMMENTED** [review]
Greet job 👏

**Review by ALiyASSER0 — APPROVED** [review]
Thanks, guys

File: Personal-Finance-Tracker/src/main/kotlin/datasource/InMemoryTransactionDataSourceImpl.kt [hunk 2]
```diff
@@ -5,13 +5,13 @@ import models.Transaction
...
 class InMemoryTransactionDataSourceImpl:TransactionDataSource {
+    val transactionList = mutableListOf<Transaction>()
     override fun createTransaction(transaction: Transaction) {
         TODO("Not yet implemented")
     }
 
-    override fun removeTransaction(transaction: Transaction) {
-        TODO("Not yet implemented")
-    }
+    override fun removeTransaction(transaction: Transaction)=
```
Comment (AbdElrahman-Rafaat-Amer) [inline Personal-Finance-Tracker/src/main/kotlin/datasource/InMemoryTransactionDataSourceImpl.kt:14]: U can use transactionList.remove(transaction) directly and this function remove the item & return true if the item in the list and return false if the item not in the list

**Review by DinaOthman21 — APPROVED** [review]
